torch/_inductor/compile_fx.py

```diff
@@ -583,126 +675,87 @@ def compile_fx_inner(
         return compiled_graph
 
     if cudagraphs:
-        # output args are tuple of first argument
-        output = output_node(gm)
-        assert len(output.args) == 1
-        stack_traces = [
-            (arg.stack_trace if isinstance(arg, torch.fx.node.Node) else None)
-            for arg in output.args[0]
-        ]
-
-        complex_memory_overlap_inputs = any(
-            complex_memory_overlap(t)
-            for t in example_inputs
-            if isinstance(t, torch.Tensor)
-        )
-
-        if not config.triton.cudagraph_support_input_mutation:
-            # Skip supports for cudagraph-managed tensors
-            from torch._inductor.cudagraph_utils import (
-                check_for_mutation_ignore_cuda_graph_managed_tensor,
-            )
-
-            has_mutation_str = check_for_mutation_ignore_cuda_graph_managed_tensor(
-                gm, compiled_graph, static_input_idxs
-            )
-            has_mutation = has_mutation_str is not None
-
-            if has_mutation:
-                compiled_graph.disabled_cudagraphs_reason = has_mutation_str
+        # check cudagraph disabling reasons from inductor lowering
+        if compiled_graph.disabled_cudagraphs_reason:
```

Comment:
I moved this if statement down one and into the cudagraphs section so all the stuff dealing with cudagraphs can be in the same place